sinks/lightstep/lightstep.go

```diff
@@ -184,11 +188,21 @@ func (ls *LightStepSpanSink) Ingest(ssfSpan *ssf.SSFSpan) error {
 		service = "unknown"
 	}
 
-	// Protect mutating the service count with a mutex
-	ls.mutex.Lock()
-	defer ls.mutex.Unlock()
-	ls.serviceCount[service]++
+	count, ok := ls.serviceCount.Load(service)
+	if !ok {
+		// ensure the value is in the map
+		// we only do this if the value was not found in the map once already, to save an
+		// allocation and more expensive operation in the typical case
```

Comment:
haha, oh to have something as useful as `Entry.or_insert_with` here (:

Reply:
We *do* - the problem is just that calling the function requires passing in the address (thereby making an allocation), even though the allocation will be unnecessary for all but the first call on each service.

For other codepaths, I wouldn't bother, except I *just* spent a large chunk of the last week removing unnecessary allocations from this path, so I'd rather not add one back in, even if it's a small one. :)

Reply:
Oh, I should clarify: the reason it's wonky is because of the hack we're doing using sync.Map and atomic variables, by which we don't actually use the map's mutex to modify the variables, because we can ensure that we only ever touch them using atomic operations.

Reply:
All that makes sense! (Just for context, in rust, the `or_insert_with` method uses a function to get the value to set in that place, which would help us, I believe (-:)